alf/algorithms/generator.py

```diff
@@ -161,18 +159,17 @@ def _predict(self, inputs, batch_size=None, training=True):
             assert self._input_tensor_spec is None
             assert batch_size is not None
         else:
-            tf.nest.assert_same_structure(inputs, self._input_tensor_spec)
-            batch_size = tf.shape(tf.nest.flatten(inputs)[0])[0]
-        shape = common.concat_shape([batch_size], [self._noise_dim])
-        noise = tf.random.normal(shape=shape)
+            nest.assert_same_structure(inputs, self._input_tensor_spec)
+            batch_size = nest.flatten(inputs)[0].shape[0]
```

Comment:
Use get_nest_batch_size for efficiency (no flatten is needed)

Reply:
updated